tlatools/org.lamport.tlatools/test/util/ParserInterface.java

```diff
@@ -0,0 +1,108 @@
+package util;
```

Comment:
This class seems largely redundant to TLCBreakpointExpression?  Can this class be moved out of the test package and be used by TLCBreakpointExpression?

Reply:
I agree it is redundant; my sort of long-term goal is to develop this API class according to the needs of the test code, and then eventually introduce it in a `tla2sany.api` java package once it has been sufficiently refined. Ideally it would then not change very much. However a few things remain before I think it's fully baked:
1. Remove `SemanticNode.error` and `Context.initialContext` global static state
2. Figure out a good module resolution process
3. Figure out a good incremental parse API, allowing users to add modules & definitions and hopefully ghost expressions to be evaluated at an arbitrary point in the tree.
4. Figure out how the API would integrate with a systematic error reporting system as sketched in https://github.com/tlaplus/rfcs/issues/15

Reply:
Let’s consider this out of scope for this PR.